contrib/docker-integration/run_multiversion.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+# Run the integration tests with multiple versions of the Docker engine
+
+set -e
+
+# Don't use /tmp because this isn't available in boot2docker
+tmpdir_template="`pwd`/docker-versions.XXXXX"
+tmpdir=`mktemp -d "$tmpdir_template"`
+trap "rm -rf $tmpdir" EXIT
+
+# If DOCKER_VOLUME is unset, create a temporary directory to cache containers
+# between runs
+# Only do this on Linux, because using /var/lib/docker from a host volume seems
+# problematic with boot2docker.
+if [ "$DOCKER_VOLUME" = "" -a `uname` = "Linux" ]; then
+	volumes_template="`pwd`/docker-versions.XXXXX"
+	volume=`mktemp -d "$volumes_template"`
+	trap "rm -rf $tmpdir $volume" EXIT
+else
+	volume="$DOCKER_VOLUME"
+fi
+
+# Released versions
+
+versions="1.6.0 1.7.0"
+
+for v in $versions; do
+	echo "Downloading Docker $v"
+	binpath="$tmpdir/docker-$v"
+	curl -L -o "$binpath" "https://test.docker.com/builds/Linux/x86_64/docker-$v"
```

Comment:
That's a good idea. It means the script wouldn't have to waste bandwidth downloading each version every time, because docker will only pull images when it needs to.

I assume we'd have to continue downloading the experimental build from https://experimental.docker.com/builds/, though.